### ISSUE TITLE
Solana provider renderer crash

### ISSUE THREAD

**Issue by darkdh** [brave/brave-browser#28147]
## Description 

@BrendanEich reported the webview is crashing whenever loading. Looking at the crashes/stack, this regressed due to https://github.com/brave/brave-core/pull/16868. Crashes:

* `da590600-668b-8709-0000-000000000000`
* `795a0600-668b-8709-0000-000000000000`
* `7a5a0600-668b-8709-0000-000000000000`
* `a15e0600-668b-8709-0000-000000000000`
* `b25e0600-668b-8709-0000-000000000000`

## Steps to Reproduce

1. install `1.49.75 Chromium: 110.0.5481.52`
2. once installed, enable wallet via `brave://wallet` (don't worry about saving the phrase, just skip)
3. once your wallet is created visit `youtube.com` and you should eventually see webview crash
4. visiting https://www.reddit.com also reproduces the same crash as visiting https://www.youtube.com

Some other websites constantly crashing:

* https://diablo2.io/base/#filter=
* https://traderie.com/diablo2resurrected?prop_Platform=PC&prop_Ladder=true
* https://steamdb.info
* https://etherscan.io

Note: On desktop, clicking on the `Wallet` icon instantly crashes YT as well.

## Actual result:

**`Example`** | **`Example`** | **`Example`** 
---------------|----------------|---------------
![image](https://user-images.githubusercontent.com/2602313/215617456-e2346184-cd1d-4212-bc1c-24245907a729.png) | ![image](https://user-images.githubusercontent.com/2602313/215617508-b59fcff6-9a87-4a48-93d0-fc62a06f976a.png) | ![image](https://user-images.githubusercontent.com/2602313/215619229-7bafbe6f-ef9d-481f-b284-24753523a7ec.png)

```
[ 00 ] brave_wallet::JSSolanaProvider::OnIsSolanaKeyringCreated(bool) ( render_frame_impl.cc:2299 )
[ 01 ] brave_wallet::JSSolanaProvider::OnIsSolanaKeyringCreated(bool) ( js_solana_provider.cc:1000 )
[ 02 ] network::mojom::CookieManager_DeleteCanonicalCookie_ForwardToCallback::Accept(mojo::Message*) ( callback.h:152 )
[ 03 ] mojo::InterfaceEndpointClient::HandleValidatedMessage(mojo::Message*) ( interface_endpoint_client.cc:1002 )
[ 04 ] mojo::internal::MultiplexRouter::Accept(mojo::Message*) ( message_dispatcher.cc:43 )
[ 05 ] mojo::MessageDispatcher::Accept(mojo::Message*) ( message_dispatcher.cc:43 )
[ 06 ] base::internal::Invoker<base::internal::BindState<void (mojo::Connector::)(unsigned int), base::internal::UnretainedWrapper<mojo::Connector, base::RawPtrBanDanglingIfSupported>>, void (unsigned int)>::Run(base::internal::BindStateBase, unsigned int) ( connector.cc:542 )
[ 07 ] base::internal::Invoker<base::internal::BindState<void (mojo::SimpleWatcher::)(int, unsigned int, mojo::HandleSignalsState const&), base::WeakPtr<mojo::SimpleWatcher>, int, unsigned int, mojo::HandleSignalsState>, void ()>::RunOnce(base::internal::BindStateBase) ( callback.h:333 )
[ 08 ] non-virtual thunk to base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWork() ( callback.h:152 )
[ 09 ] base::MessagePumpCFRunLoopBase::RunWork() ( message_pump_mac.mm:475 )
[ 10 ] base::mac::CallWithEHFrame(void () block_pointer)
[ 11 ] base::MessagePumpCFRunLoopBase::RunWorkSource(void*) ( message_pump_mac.mm:447 )
[ 12 ] 0x1ac7a9a30
[ 13 ] 0x1ac7a99c4
[ 14 ] 0x1ac7a9734
[ 15 ] 0x1ac7a8338
[ 16 ] 0x1ac7a78a0
[ 17 ] 0x1ad6afe54
[ 18 ] base::MessagePumpNSRunLoop::DoRun(base::MessagePump::Delegate*) ( message_pump_mac.mm:768 )
[ 19 ] base::MessagePumpCFRunLoopBase::Run(base::MessagePump::Delegate*) ( message_pump_mac.mm:172 )
[ 20 ] base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::Run(bool, base::TimeDelta) ( thread_controller_with_message_pump_impl.cc:644 )
[ 21 ] base::RunLoop::Run(base::Location const&) ( run_loop.cc:0 )
[ 22 ] content::RendererMain(content::MainFunctionParams) ( renderer_main.cc:330 )
[ 23 ] content::RunOtherNamedProcessTypeMain(std::Cr::basic_string<char, std::Cr::char_traits<char>, std::Cr::allocator<char>> const&, content::MainFunctionParams, content::ContentMainDelegate*) ( content_main_runner_impl.cc:746 )
[ 24 ] content::ContentMainRunnerImpl::Run() ( content_main_runner_impl.cc:1100 )
[ 25 ] content::RunContentProcess(content::ContentMainParams, content::ContentMainRunner*) ( content_main.cc:344 )
[ 26 ] content::ContentMain(content::ContentMainParams) ( content_main.cc:372 )
[ 27 ] ChromeMain ( chrome_main.cc:174 )
[ 28 ] main ( chrome_exe_main_mac.cc:216 )
[ 29 ] 0x1ac39fe4c
```

## Expected result:

Brave should be crashing when loading websites. Specifically shouldn't be crashing YT.

## Reproduces how often: 

100% reproducible using the STR/Cases outlined above.

## Desktop Brave version:

```
Brave | 1.49.75 Chromium: 110.0.5481.52 (Official Build) nightly (64-bit)
-- | --
Revision | 979113183ded4544a2c443aceb1629c430907e52-refs/branch-heads/5481@{#636}
OS | Windows 11 Version 22H2 (Build 22621.1105)
```

## Android Device details:

   - Install type (ARM, x86): `ARM`
   - Device type (Phone, Tablet, Phablet): `Pixel 6`
   - Android version: `Android 13`

## Version/Channel Information:

- Can you reproduce this issue with the current release? `N/A`
- Can you reproduce this issue with the beta channel? `Yes` - will require builds >= `1.48.149`
- Can you reproduce this issue with the nightly channel? `Yes` - Reproduced using `1.49.76 Chromium: 110.0.5481.52`

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? `N/A`
- Does the issue resolve itself when disabling Brave Rewards? `N/A`
- Is the issue reproducible on the latest version of Chrome? `N/A`

**Comment by Uni-verse** [comment]
Reproduced using version `1.49.75 Chromium 110.0.5481.52` on `Samsung Galaxy S21 5G` running `Android 13`

STR: 

1. Install Nightly
2. Create brave wallet
3. Open new tab, navigate to `youtube.com` and play any video

<img width="500" alt="0" src="https://user-images.githubusercontent.com/17885425/215619024-77e75a41-6cd7-489b-9175-446b1acca00a.png">

**Comment by kjozwiak** [comment]
The above requires `1.47.190` or higher for `1.47.x` verification 👍

**Comment by kjozwiak** [comment]
Moving this into `1.48.x` as we neve received another `C109` minor chromium bump that would have triggered another `1.47.x` release. We'll leave https://github.com/brave/brave-core/milestone/267 as there's code that landed in `1.47.x` but we'll close off the `1.47.x - Release #6` milestone and move all the issues within `1.48.x`.

The above can be verified with `1.48.156` for `1.48.x` as it's been merged into a `1.48.x` a while ago as per https://github.com/brave/brave-core/pull/16937.

**Comment by srirambv** [comment]
Verification passed on 

Brave | 1.48.157 Chromium: 110.0.5481.77 (Official Build) (64-bit)
-- | --
Revision | `65ed616c6e8ee3fe0ad64fe83796c020644d42af-refs/branch-heads/5481@{#839}`
OS | Windows 11 Version 22H2 (Build 22621.1105)

- Verified steps from issue description 
- Verified creating wallet and visiting the following links doesn't crash the browser or renderer
   - `https://www.youtube.com/`
   - `https://www.reddit.com/`
   - `https://diablo2.io/base/#filter=`
   - `https://traderie.com/diablo2resurrected?prop_Platform=PC&prop_Ladder=true`
   - `https://steamdb.info/`
   - `https://etherscan.io/`
- Verified no webview crash on the above pages when wallet is created after loading the sites
- Verified no webview crash / browser crash when clicking on the wallet icon
- Verified no webview crash / browser crash when triggering wallet icon on a Dapp with the above sites being loaded in other tabs


--------------
Verification passed on Oppo Reno 5 with Android 13 running 1.48.157
- Verified steps from issue description 
- Verified creating wallet and visiting the following links doesn't crash the browser or renderer
   - `https://www.youtube.com/`
   - `https://www.reddit.com/`
   - `https://diablo2.io/base/#filter=`
   - `https://traderie.com/diablo2resurrected?prop_Platform=PC&prop_Ladder=true`
   - `https://steamdb.info/`
   - `https://etherscan.io/`
- Verified no webview crash on the above pages when wallet is created after loading the sites
- Verified no webview crash / browser crash when clicking on the wallet option from menu
- Verified no webview crash / browser crash when triggering wallet icon on a Dapp with the above sites being loaded in other tabs
